osc-server/src/main/java/org/osc/core/broker/util/crypto/X509TrustManagerFactory.java

```diff
@@ -220,13 +249,36 @@ private void reloadTrustManager() throws Exception {
 
     @Override
     public void addEntry(File file) throws Exception {
-        CertificateFactory cf = CertificateFactory.getInstance("X.509");
         String newAlias = cleanFileName(FilenameUtils.removeExtension(file.getName()));
+        CertificateFactory cf = CertificateFactory.getInstance("X.509");
         X509Certificate certificate;
         try (FileInputStream inputStream = new FileInputStream(file)) {
             certificate = (X509Certificate) cf.generateCertificate(inputStream);
         }
+        setOrUpdateCertificate(certificate, newAlias);
+    }
+
+    @Override
+    public void addEntry(X509Certificate certificate, String newAlias) throws Exception {
+        if (fingerprintNotExist(getSha1Fingerprint(certificate))) {
```

Comment:
SHA1 is no longer used and it is deprecated. Do we create issue to enhance it to SHA256?

Reply:
Yes, we should